frontend/src/components/Modal/ProfileModal/ProfileModal.tsx

```diff
@@ -0,0 +1,160 @@
+        {
+            name: ProfileInputType.ProfileName,
+            id: "profileName",
+            label: "Profile name",
+            placeholder: "Profile name",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.ProfilePicture,
+            id: "profilePicture",
+            label: "Profile picture",
+            placeholder: "Profile picture URL",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.CoverPicture,
+            id: "coverPicture",
+            label: "Cover picture",
+            placeholder: "Cover picture URL",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.About,
+            id: "about",
+            label: "About",
+            placeholder: "Tell something about you",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.WebsiteUrl,
+            id: "websiteUrl",
+            label: "Website address",
+            placeholder: "Website URL",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.FacebookUrl,
+            id: "facebookUrl",
+            label: "Facebook profile",
+            placeholder: "Facebook profile URL",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.InstagramUrl,
+            id: "instagramUrl",
+            label: "Instagram profile",
+            placeholder: "Instagram profile URL",
+            type: "text",
+        },
+        {
+            name: ProfileInputType.TwitterUrl,
+            id: "twitterUrl",
+            label: "Twitter profile",
+            placeholder: "Twitter profile URL",
+            type: "text",
+        },
+    ];
+
+    const [response, setResponse] = useState<string | null>(null);
+
+    const hideMessage = () => {
+        setTimeout(() => {
+            setResponse(null);
+        }, 2000);
+    };
+    const init = isNew
+        ? {
+              profileName: "",
+              profilePicture: "",
+              coverPicture: "",
+              about: "",
+              websiteUrl: "",
+              instagramUrl: "",
+              twitterUrl: "",
+              facebookUrl: "",
+          }
+        : {
+              profileName: activeProfile?.profileName,
+              profilePicture: activeProfile?.profilePicture,
+              coverPicture: activeProfile?.coverPicture,
+              about: activeProfile?.about,
+              websiteUrl: activeProfile?.websiteUrl,
+              instagramUrl: activeProfile?.instagramUrl,
+              twitterUrl: activeProfile?.twitterUrl,
+              facebookUrl: activeProfile?.facebookUrl,
+          };
+    const formik: FormikValues = useFormik({
+        initialValues: init,
+        validationSchema,
+        validateOnChange: false,
+        onSubmit: async (values) => {
+            setResponse(null);
+            isNew
+                ? await addProfile(values, userID).then((data) => {
```

Comment:
suggestion: you can create a function that handles the response in `then`, since both functions do the same thing.